starknet_devnet/blueprints/rpc/misc.py

```diff
@@ -27,12 +78,48 @@ async def syncing() -> Union[dict, bool]:
     return False
 
 
-# pylint: disable=redefined-builtin
-async def get_events(filter: dict) -> dict:
+# pylint: disable=too-many-arguments
+async def get_events(
+    from_block: BlockId,
+    to_block: BlockId,
+    address: Address = "",
+    keys: List[Address] = Empty,
+    chunk_size: int = 0,
+    continuation_token: str = "",
+) -> str:
     """
-    Returns all events matching the given filter
+    Returns all events matching the given filters.
+
+    In our implementation continuation_token is just a number.
+
+    In state.starknet_wrapper.get_state().events there is no relation between blocks.
+    This is why we need to iterate block by block, take all events,
+    and chunk it later which is not an optimal solution.
     """
-    raise NotImplementedError()
+    events = []
+    keys = [] if keys == Empty else [int(k, 0) for k in keys]
```

Comment:
When comparing with singletons, the `is` operator should be used in favor of `==`: https://stackoverflow.com/a/14247419

Reply:
fixed